src/sinks/new_relic/model.rs

```diff
@@ -57,39 +41,66 @@ impl TryFrom<Vec<Event>> for MetricsApiModel {
 
         for buf_event in buf_events {
             if let Event::Metric(metric) = buf_event {
-                // Future improvement: put metric type. If type = count, NR metric model requires an interval.ms field, that is not provided by the Vector Metric model.
-                match metric.value() {
-                    MetricValue::Gauge { value } => {
-                        metric_array.push((
-                            Value::from(metric.name().to_owned()),
-                            Value::from(
-                                NotNan::new(*value).map_err(|_| {
-                                    NewRelicSinkError::new("NaN value not supported")
-                                })?,
-                            ),
-                            Value::from(metric.timestamp()),
-                        ));
-                    }
-                    MetricValue::Counter { value } => {
-                        metric_array.push((
-                            Value::from(metric.name().to_owned()),
-                            Value::from(
-                                NotNan::new(*value).map_err(|_| {
-                                    NewRelicSinkError::new("NaN value not supported")
-                                })?,
-                            ),
-                            Value::from(metric.timestamp()),
-                        ));
+                // Generate Value::Object() from BTreeMap<String, String>
+                let (series, data, _) = metric.into_parts();
+                let attr = series.tags.map(|tags| {
+                    Value::from(
+                        tags.iter_single()
+                            .map(|(key, value)| (key.to_string(), Value::from(value)))
+                            .collect::<BTreeMap<_, _>>(),
+                    )
+                });
```

Comment:
I would suggest moving this down to where `attr` is actually used. This avoids doing the work to set up the attributes if they are not used, as well as allowing for simplifications of the coding.